src/telliot_core/gas/legacy_gas.py

```diff
@@ -38,3 +45,47 @@ async def ethgasstation(style: ethgastypes = "fast", retries: int = 2) -> Option
         except Exception as e:
             logger.error(f"Error fetching gas price: {e}")
     return None
+
+
+gas_station = {
+    1: GasStation(api="https://ethgasstation.info/json/ethgasAPI.json", default_speed="fast"),
+    5: GasStation(api="https://ethgasstation.info/json/ethgasAPI.json", default_speed="fast"),
+    137: GasStation(api="https://gasstation-mainnet.matic.network", default_speed="safeLow"),
+    80001: GasStation(api="https://gasstation-mainnet.matic.network", default_speed="safeLow"),
+}
+
+
+async def legacy_gas_station(chain_id: int, speed: Optional[ethgastypes] = None, retries: int = 2) -> Optional[int]:
+    """Fetch gas price from ethgasstation in gwei"""
+
+    if speed is None:
+        try:
+            speed = gas_station[chain_id].default_speed
+        except KeyError:
+            logger.error(f"Please add gas station api for chain id: {chain_id}")
+            return None
+
+    for _ in range(retries):
```

Comment:
Instead of using a for loop for number of retries, you could use some built-ins like in `telliot_feeds/sources/blockhash_aggregator.py` to do retries and handle common error codes:
```python
...

retry_strategy = Retry(
    total=3,
    backoff_factor=1,
    status_forcelist=[429, 500, 502, 503, 504],
    allowed_methods=["GET"],
)
adapter = HTTPAdapter(max_retries=retry_strategy)


def block_num_from_timestamp(timestamp: int) -> Optional[int]:
    with requests.Session() as s:
        s.mount("https://", adapter)
        try:
            rsp = s.get(
                "https://api.etherscan.io/api"
                "?module=block"
                "&action=getblocknobytime"
                f"&timestamp={timestamp}"
                "&closest=before"
                "&apikey="
            )
        except requests.exceptions.ConnectTimeout:
            logger.error("Connection timeout getting ETH block num from timestamp")
            return None
        except requests.exceptions.RequestException as e:
            logger.error(f"Etherscan API error: {e}")
            return None
 ```

Reply:
Maybe we can add this another time